ONSdigital/census-rm-case-service
# Fix null caserefs

# Motivation and Context
It was noticed that the caseRefs were null since the performance changes. A defect was raised by @AdamHawtin.

# What has changed
After testing, it seems that the default value sequence generator does not get triggered on insert unless we do a `saveAndFlush` - the `save` that was getting called was inadequate, for some reason. The code's been changed so that it will do a `saveAndFlush` which will create the primary key and the case ref, then it will carry on and do the other things, like recording the IAC code in the audit table and sending the message to the Action service.

# How to test?
Load a sample and check that the `caseRef` column has values in it on the `case` table.

# Links
Trello: https://trello.com/c/j3MG0GKI/638-defect-case-refs-are-null
Original trello card which introduced the regression defect: https://trello.com/c/cHLbOX2I/606-increase-performance-of-case-distribution
